test/functional/p2p_segwit.py

```diff
@@ -191,14 +191,13 @@ def announce_tx_and_wait_for_getdata(self, tx, success=True, use_wtxid=False):
     def announce_block_and_wait_for_getdata(self, block, use_header, timeout=60):
         with p2p_lock:
             self.last_message.pop("getdata", None)
-            self.last_message.pop("getheaders", None)
```

Comment:
this can't be removed either. (same reason as above)

Reply:
I don't think this is the same case as above:

Here we are using `wait_for_getheaders` expecting a specific block hash. Therefore, this should not be the case as long as a specific block is not announced manually and then re-announced using this method, which is not what the test is doing AFAICT

Reply:
oh good point.